joomla/joomla-cms
# [4] Add intl extension check to system info, and while there clean up deprecations/types/[]

### Summary of Changes

As a result of https://github.com/joomla/joomla-cms/pull/33451 and https://github.com/joomla/joomla-cms/pull/33454 I noticed ICONV was checked but not INTL - as both are needed for `File::makeSafe` it makes sense to have both listed in the System Information for ease of reference. 

While in this code, took the opportunity to remove deprecated `getDBO` method calls and add param types and return types. 

<img width="574" alt="Screenshot 2021-05-01 at 00 18 43" src="https://user-images.githubusercontent.com/400092/116762602-cec3be00-aa12-11eb-87b0-4c5953a53e40.png">

### Testing Instructions

Load  System Information in Joomla Admin - if it loads, great. Check PHP tab and see your status of INTL (checks for presence of the function `transliterator_transliterate` which is the one Joomla uses in `File::makeSafe`

Try JSON and TXT download.

### Actual result BEFORE applying this Pull Request

Works - but intl extension not in the list

### Expected result AFTER applying this Pull Request

Still works - but calls one less deprecated function and code is slightly more modern and less warnings in phpStorm. 

### Documentation Changes Required

None
